util/patricia_trie/src/triedb.rs

```diff
@@ -97,7 +96,10 @@ impl<'db> TrieDB<'db> {
 			Node::Extension(ref slice, ref item) => {
 				write!(f, "'{:?} ", slice)?;
 				if let Ok(node) = self.get_raw_or_lookup(&*item) {
-					self.fmt_all(Node::decoded(&node), f, deepness)?;
+					match Node::decoded(&node) {
+						Ok(n) => self.fmt_all(n, f, deepness)?,
+						Err(err) => writeln!(f, "ERROR decoding node extension Rlp: {}", err)?,
```

Comment:
unrelated:
instead of `writeln`, we should use [debug_struct](https://doc.rust-lang.org/1.5.0/std/fmt/struct.Formatter.html#method.debug_struct) which takes into an account an [alternate](https://doc.rust-lang.org/1.5.0/std/fmt/struct.Formatter.html#method.alternate) flag of the formatter which allows [pretty-printing](https://doc.rust-lang.org/std/fmt/trait.Debug.html#examples)